OpenRA.Mods.Common/UtilityCommands/ExtractTraitDocsCommand.cs

```diff
@@ -35,74 +36,58 @@ void IUtilityCommand.Run(Utility utility, string[] args)
-
-					var i = 0;
-					foreach (var require in requires)
+					Namespace = type.Namespace,
+					Name = type.Name.EndsWith("Info") ? type.Name.Substring(0, type.Name.Length - 4) : type.Name,
+					Description = string.Join(" ", type.GetCustomAttributes<DescAttribute>(false).SelectMany(d => d.Lines)),
+					RequiresTraits = RequiredTraitTypes(type)
+						.Select(y => y.Name)
+						.Select(z => z.EndsWith("Info") ? z.Substring(0, z.Length - 4) : z),
+					InheritedTypes = type.BaseTypes()
+						.Select(y => y.Name)
+						.Where(y => y != type.Name && y != $"{type.Name}Info" && y != "Object")
+						.Select(z => z.EndsWith("Info") ? z.Substring(0, z.Length - 4) : z)
+						.ToArray(),
+					Properties = FieldLoader.GetTypeLoadInfo(type).Select(fi => new
 					{
-						var n = require.Name;
-						var name = n.EndsWith("Info") ? n.Remove(n.Length - 4, 4) : n;
-						doc.Append($"[`{name}`](#{name.ToLowerInvariant()}){(i + 1 == reqCount ? ".\n" : ", ")}");
-						i++;
-					}
-				}
-
-				var infos = FieldLoader.GetTypeLoadInfo(t);
-				if (!infos.Any())
-					continue;
-				doc.AppendLine();
-				doc.AppendLine("| Property | Default Value | Type | Description |");
-				doc.AppendLine("| -------- | --------------| ---- | ----------- |");
-				var liveTraitInfo = Game.ModData.ObjectCreator.CreateBasic(t);
-				foreach (var info in infos)
-				{
-					var fieldDescLines = info.Field.GetCustomAttributes<DescAttribute>(true).SelectMany(d => d.Lines);
-					var fieldType = Util.FriendlyTypeName(info.Field.FieldType);
-					var loadInfo = info.Field.GetCustomAttributes<FieldLoader.SerializeAttribute>(true).FirstOrDefault();
-					var defaultValue = loadInfo != null && loadInfo.Required ? "*(required)*" : FieldSaver.SaveField(liveTraitInfo, info.Field.Name).Value.Value;
-					doc.Append($"| {info.YamlName} | {defaultValue} | {fieldType} | ");
-					foreach (var line in fieldDescLines)
-						doc.Append(line + " ");
-					doc.AppendLine("|");
-				}
-			}
+						PropertyName = fi.YamlName,
+						DefaultValue = FieldSaver.SaveField(objectCreator.CreateBasic(type), fi.Field.Name).Value.Value,
+						Type = Util.FriendlyTypeName(fi.Field.FieldType),
+						Description = string.Join(" ", fi.Field.GetCustomAttributes<DescAttribute>(true).SelectMany(d => d.Lines)),
+						OtherAttributes = fi.Field.CustomAttributes
+							.Where(a => a.AttributeType.Name != nameof(DescAttribute) && a.AttributeType.Name != nameof(FieldLoader.LoadUsingAttribute))
+							.Select(a =>
+							{
+								var name = a.AttributeType.Name;
+								name = name.EndsWith("Attribute") ? name.Substring(0, name.Length - 9) : name;
+
+								return new
+								{
+									Name = name,
+									Value = a.ConstructorArguments.Select(b => b.Value)
+								};
+							})
+					})
+				});
+
+			var result = new
+			{
+				Version = version,
+				TraitInfos = traitTypesInfo
+			};
 
-			Console.Write(doc.ToString());
+			return Newtonsoft.Json.JsonConvert.SerializeObject(result);
```

Comment:
Can we please have a `using Newtonsoft.Json` above and then `return JsonConvert.SerializeObject(result, Formatting.Indented);` for these?